### ISSUE TITLE
statistics service causes horizon rate limiting

### ISSUE THREAD

**Issue by robvanmieghem** [threefoldfoundation/tft-stellar#361]
response from horizon:
```
stellar_sdk.exceptions.BadRequestError: {
"type": "https://stellar.org/horizon-errors/rate_limit_exceeded",
"title": "Rate Limit Exceeded",
"status": 429,
"detail": "The rate limit for the requesting IP address is over its alloted limit. The allowed limit and requests left per time period are communicated to clients via the http response headers 'X-RateLimit-*' headers."
}
```
requests on testserver:
```
27.0.0.1 - - [2021-04-15 08:24:21] "GET /api/stats?detailed=true HTTP/1.0" 200 137 319.200234
127.0.0.1 - - [2021-04-15 08:24:28] "GET /api/stats?detailed=true HTTP/1.0" 200 137 326.798744
127.0.0.1 - - [2021-04-15 08:24:31] "GET /api/stats?detailed=true HTTP/1.0" 200 137 329.399293
127.0.0.1 - - [2021-04-15 08:24:32] "GET /api/stats?detailed=true HTTP/1.0" 200 137 330.423832
127.0.0.1 - - [2021-04-15 08:24:33] "GET /api/stats?detailed=true HTTP/1.0" 200 137 331.140167
127.0.0.1 - - [2021-04-15 08:24:35] "GET /api/stats?detailed=true HTTP/1.0" 200 137 333.439018
127.0.0.1 - - [2021-04-15 08:24:38] "GET /api/stats?detailed=true HTTP/1.0" 200 137 336.920308
127.0.0.1 - - [2021-04-15 08:24:39] "GET /api/stats?detailed=true HTTP/1.0" 200 137 337.346061
127.0.0.1 - - [2021-04-15 08:25:01] "GET /api/stats?detailed=true HTTP/1.0" 200 4213 0.000808
127.0.0.1 - - [2021-04-15 08:25:01] "GET /api/stats?detailed=true HTTP/1.0" 200 4213 0.000768
127.0.0.1 - - [2021-04-15 08:25:01] "GET /api/stats?detailed=true HTTP/1.0" 200 4213 0.000829
127.0.0.1 - - [2021-04-15 08:25:02] "GET /api/stats?detailed=true HTTP/1.0" 200 4213 0.000730
127.0.0.1 - - [2021-04-15 08:26:02] "GET /api/stats?detailed=true HTTP/1.0" 200 4213 0.000652
```

they all come from the same ip:
```
34.227.238.95 - - [15/Apr/2021:06:34:01 +0000] "GET /stellar_stats/api/stats?detailed=true HTTP/1.1" 200 4076 "-" "-" 106 0.031 [default-tftstatistics-443] [] 10.244.10.191:443 4076 0.032 200 872e0f6f80dd9608a09d15a2490e3925
34.227.238.95 - - [15/Apr/2021:06:34:01 +0000] "GET /stellar_stats/api/stats?detailed=true HTTP/1.1" 200 4076 "-" "-" 106 0.038 [default-tftstatistics-443] [] 10.244.10.191:443 4076 0.040 200 39a73a8381a95ff370fe9fcb65cc524b
34.227.238.95 - - [15/Apr/2021:06:34:01 +0000] "GET /stellar_stats/api/stats?detailed=true HTTP/1.1" 200 4076 "-" "-" 106 0.034 [default-tftstatistics-443] [] 10.244.10.191:443 4076 0.036 200 6d02868ede73031a1b536104a83b579c
```

Since if the cache is expired and the request is done multiple times in parallel, a lot of queries are made against the horizon servers before the cache is filled up again.

This not only impacts the statistics service but all services ( like transaction funding) that use the horizon servers.

**Comment by robvanmieghem** [comment]
@abdulgig are you doing these parallell request?

**Comment by abdulgig** [comment]
The cause/reasons have been discussed. Probably happened when developing the foundation accounts page. The statistics API endpoint needs to have checks to discard any requests exceeding the threshold to prevent this

**Comment by robvanmieghem** [comment]
it needs to internally serialize filling the cache using a gevent pool

**Comment by robvanmieghem** [comment]
With the new implementation of the dashboard, it only gets fetched once a day so it can be postponed

**Comment by robvanmieghem** [comment]
Will be properly soved by #401
